src/net/if_.rs

```diff
@@ -0,0 +1,48 @@
+//! Network interface name resolution.
+//!
+//! Uses Linux and/or POSIX functions to resolve interface names like "eth0"
+//! or "socan1" into device numbers.
+
+use libc::{c_char, c_uint};
+use std::ffi::{CString, NulError};
+use std::io;
+
+/// Error that can occur during interface name resolution.
+#[derive(Debug)]
+pub enum NameToIndexError {
+    /// Failed to allocate a C-style string to for the syscall
+    NulError,
+    IOError(io::Error),
+}
```

Comment:
A fourth option is now in 71c6836: Just returned Error::invalid_argument() if a string containig a null-byte is passed.